src/Cli/dotnet/commands/dotnet-workload/install/NetSdkManagedInstaller.cs

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.NuGetPackageInstaller;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ToolPackage;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using NuGet.Versioning;
+using static Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver;
+using EnvironmentProvider = Microsoft.DotNet.NativeWrapper.EnvironmentProvider;
+
+namespace Microsoft.DotNet.Workloads.Workload.Install
+{
+    internal class NetSdkManagedInstaller : PackWorkloadInstallerBase
+    {
+        private readonly IReporter _reporter;
+        private readonly string _workloadMetadataDir;
+        private readonly string _installedWorkloadDir = "InstalledWorkloads";
+        private readonly string _installedPacksDir = "InstalledPacks";
+        protected readonly string _dotnetDir;
+        protected readonly string _tempPackagesDir;
+        private INuGetPackageInstaller _nugetPackageInstaller;
+
+        public NetSdkManagedInstaller(
+            IReporter reporter,
+            INuGetPackageInstaller nugetPackageInstaller = null,
+            string dotnetDir =  null)
+        {
+            _dotnetDir = dotnetDir ?? EnvironmentProvider.GetDotnetExeDirectory();
+            _tempPackagesDir = Path.Combine(_dotnetDir, "metadata", "temp");
+            _nugetPackageInstaller = nugetPackageInstaller ?? new NuGetPackageInstaller(_tempPackagesDir, sourceUrl: "https://pkgs.dev.azure.com/azure-public/vside/_packaging/xamarin-impl/nuget/v3/index.json");
+            _workloadMetadataDir = Path.Combine(_dotnetDir, "metadata", "workloads");
+            _reporter = reporter;
+        }
+
+        public override void InstallWorkloadPack(PackInfo packInfo, string featureBand, bool useOfflineCache = false)
+        {
+            if (useOfflineCache)
+            {
+                throw new NotImplementedException();
```

Comment:
make sure todo and NotImplementedException is refected in the github issues. So we don't miss them later

Reply:
Sure, I'll keep https://github.com/dotnet/sdk/issues/16483 up to date with these items